friends/src/actions/index.js

```diff
@@ -0,0 +1,47 @@
+import axios from "axios";
+export const LOGIN_START = "LOGIN_START";
+export const LOGIN_SUCCESS = "LOGIN_SUCCESS";
+export const LOGIN_FAILURE = "LOGIN_FAILURE";
+
+export const LOGOUT = "LOGOUT";
+
+export const FRIENDS_START = "FRIENDS_START";
+export const FRIENDS_SUCCESS = "FRIENDS_SUCCESS";
+export const FRIENDS_FAILURE = "FRIENDS_FAILURE";
+
+export const ADD_START = "ADD_START";
+export const ADD_SUCCESS = "ADD_SUCCESS";
+export const ADD_FAILURE = "ADD_FAILURE";
+
+const baseUrl = "http://localhost:5000";
+
+export const handleLogin = user => dispatch => {
+  dispatch({ type: LOGIN_START });
+  axios
+    .post(`${baseUrl}/api/login`, user)
+    .then(res => dispatch({ type: LOGIN_SUCCESS, payload: res.data.payload }))
```

Comment:
this would be a good place to set the returned token to localStorage
`
 The login action creator should dispatch a "logging in" action, return the promise created by axios.post, then save the returned token to localStorage. 
`

Reply:
Check TK 
 - https://learn.lambdaschool.com/fsw/module/recQlbCZf9tOVgiXC/

```
export const login = creds => dispatch => {
  dispatch({ type: LOGIN_START });
  return axios
    .post('/api/login', creds)
    .then(res => {
      localStorage.setItem('token', res.data.token);
      dispatch({ type: LOGIN_SUCCESS, payload: res.data });
    })
    .catch(err => {
      dispatch({ type: LOGIN_FAILURE, payload: err.response.message });
    });
};

```